xdsl/utils/exceptions.py

```diff
@@ -71,6 +71,9 @@ class ParseError(Exception):
     def __repr__(self) -> str:
         return self.span.print_with_context(self.msg)
 
+    def with_context(self) -> str:
+        return self.span.print_with_context(self.msg)
+
```

Comment:
Genuine question, it's the exact duplicate of `__repr__`, can't `__repr__` be used instead?
Otherwise make __repr__ use this one?

Reply:
Not sure why exactly I chose that way, I guess I didn't want to depend on the current behavior of `__repr__` (as it conflicts with the stated goal of `__repr__`)...